### PR TITLE
Apache Commons Math, Dead Reckoning etc.

### PR DESCRIPTION
**Pull request by camejia**
Looks neat other than the weird missing lib. That's odd.

### CODE REVIEW

**Comment by camejia** [comment]
Ugh, Travis CI doesn't seem to be finding Apache Commons Math even though I added it to pom.xml.  My tests pass on my local computer.  I may be using GitHub incorrectly.  Any suggestions?

**Comment by leif81** [comment]
I think you'll want to remove the line `<scope>test</scope>` from the `pom.xml`. What that's useful for is when you want to make a dependency only available for unit tests.

**Comment by leif81** [comment]
Also, if you're up to it, it sometimes helps to put all the related commits onto one branch. That way you can avoid the merge commits appearing in your pull request which can sometimes make things a little odd to review.

**Comment by leif81** [comment]
I meant to mention, let me know if you need a hand with that, it's sometimes a bit daunting if you haven't done a rebase before.

**Comment by camejia** [comment]
Removing the test scope from pom.xml seems to have fixed Travis CI on one of my pull requests, but not on the other.  I don't know why I have two pull requests.  Unfortunately I don't understand git or GitHub, and I'm surprised that I've gotten as far as I have by just clicking buttons.  Any help straightening this out would be appreciated (and if it's all stuff that can be done from the GitHub website or GitHub desktop, that would be great).

P.S. I tried adding commons-math3-3.6.jar in the lib directory, but maybe that wasn't necessary?

**Comment by leif81** [comment]
Ya the `lib` folder isn't neccessary for the Travis build to pass. Travis will use Maven, so the dependencies it fetches are defined in the `pom.xml`. The `lib` folder is leftover from the old Ant build system.

I think what happened is that your local forked repository is missing some of the work that's been merged to `master` recently. This is kind of a common github issue. Here's the official github documentation on the subject https://help.github.com/articles/syncing-a-fork/. The GitHub desktop client doesn't make it much easier yet.

I personally prefer and recommend using *rebase* when doing it. Try this:

```
$ git remote add upstream https://github.com/open-dis/open-dis-java.git
$ git checkout master
$ git pull --rebase upstream master
$ git push --force
```

If it works it should update this pull request.

**Comment by camejia** [comment]
Thanks for the hints, but still no luck.  I think I'll save all of my files, delete all of my pull requests, branches, and forks and try again this evening.

File: pom.xml [hunk 1]
```diff
@@ -53,6 +53,12 @@
             <version>4.12</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-math3</artifactId>
+            <version>3.6</version>
+            <scope>test</scope>
```
Comment (leif81) [inline pom.xml]: If you remove this line and I think the Travis job will pass.

**Comment by leif81** [comment]
This is really close now I think Chris. 

I believe all that's missing is the fix to the pom.xml to remove the `<scope>test</scope>`. If you fix that and make a push this PR will update and I suspect the Travis build will pass.

**Comment by camejia** [comment]
I made this quick edit directly on the GitHub website and now all checks pass.  Finally!  Do I need to do something else?  I don't know what a "push" is.  Thanks for the help, Leif.  It's not clear to me whether when all this is done if there will still be commons-math3-3.6.jar in the lib directory, but maybe that can be cleared up later.

**Comment by leif81** [comment]
Nice work Chris.

At a glance this looks pretty fantastic. You've done a *lot* of work here. The change in terms of lines of code is very large, but the unit tests you added are very nice reassurance.

@mcgredonps what do you think?

**Pull request merged**